include/bluetooth/crypto.h

```diff
@@ -67,6 +67,55 @@ int bt_encrypt_le(const u8_t key[16], const u8_t plaintext[16],
 int bt_encrypt_be(const u8_t key[16], const u8_t plaintext[16],
 		  u8_t enc_data[16]);
 
+
+/** @brief Decrypt big-endian data with AES-CCM.
+ *
+ *  Decrypts and authorizes @c enc_data with AES-CCM, as described in
+ *  https://tools.ietf.org/html/rfc3610.
+ *
+ *  Assumes that the MIC follows directly after the encrypted data.
+ *
+ *  @param key       128 bit MS byte first key
+ *  @param nonce     13 byte MS byte first nonce
+ *  @param enc_data  Encrypted data
+ *  @param len       Length of the encrypted data
+ *  @param aad       Additional input data
+ *  @param aad_len   Additional input data length
+ *  @param plaintext Plaintext buffer to place result in
+ *  @param mic_size  Size of the trailing MIC (in bytes)
+ *
+ *  @retval 0        Successfully decrypted the data.
+ *  @retval -EINVAL  Invalid parameters.
+ *  @retval -EBADMSG Authentication failed.
+ */
+int bt_ccm_decrypt(const u8_t key[16], u8_t nonce[13], const u8_t *enc_data,
```

Comment:
Would it be sensible to add `_be` at the end here?
I'm thinking mostly in the case we would ever need an `_le` version here. But it would also be consistent with the other encrypt functions here.

Reply:
AFAIK big-endian is some kind of de-facto standard for crypto APIs. We have _le and _be for the AES-128 block-cipher since this makes for a more efficient implementation when the protocol data is little endian and the underlying hardware cipher is also little endian (I believe this to be the case on nRF?). That said, if you guys foresee the need for a little-endian CCM variant then I'm not opposed to it.

Reply:
I don't foresee it no. And also we could just add _le in that case so no problem.

The nRF ECB peripheral uses BE as far as I know.